### PR TITLE
Added automatic markdown link checking and fixed broken links

### LINKED ISSUE #1969: broken link
at
https://github.com/dotnet/iot/blob/main/samples/force-sensitive-resistor/README.md
the following link is broken:
[please refer this for more about binding MCP3008](https://github.com/dotnet/iot/tree/main/src/devices/Mcp3008/samples)

### PR DESCRIPTION
**Pull request by Jimmys20**
Fixes #1969

This PR enables automatic markdown link checking using [markdown-link-check](https://github.com/tcort/markdown-link-check) utility. I fixed most of the broken links and I created an ignore list for the links I wasn't sure how to fix.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/1970)

### CODE REVIEW

**Review by pgrawehr — APPROVED** [review]

**Review by krwq — APPROVED** [review]
LGTM, +1 if you make our build break whenever there is any broken link. Possibly markdownlint has option for that: https://github.com/dotnet/iot/blob/main/.vsts-ci.yml#L107

**Comment by krwq** [comment]
FYI: https://github.com/tcort/markdown-link-check

**Comment by krwq** [comment]
You can use something similar to this to run it on all files:

```
ls -r *.md | %{markdown-link-check $_.FullName}
```

It doesn't show error code correctly like this though so will need some adjustment.

There are plenty of other dead links in the repo

**Comment by Jimmys20** [comment]
I'm not sure I will be able to do this. I'm not very experienced with yaml and powershell scripts.

**Comment by Jimmys20** [comment]
This is the list of the problematic links according to markdown-link-check that I wasn't sure how to fix:

- src\devices\Amg88xx\README.md
  - https://eu.industrial.panasonic.com/sites/default/pidseu/files/downloads/files/

- src\devices\Ak8963\README.md
  - https://www.akm.com/akm/en/file/datasheet/AK8963C.pdf

- src\devices\README.md
  - https://dotnet.myget.org/feed/dotnet-core/package/nuget/Iot.Device.Bindings

- samples\bmp280-sensor-azure-iot-hub\Readme.md
  - ../.vscode/launch.json
  - ../.vscode/tasks.json

- Documentation\creating-new-release.md
  - https://dev.azure.com/dnceng/internal/_build?definitionId=239

- Documentation\README.md
  - https://www.pine64.org/?page_id=46823

- src\devices\Media\VideoDevice\samples\MjpegStream\README.md
  - http://url/image/settimezone?timezone=1
  - http://url/image
  - http://url/image/stream

- src\devices\Pca95x4\README.md
  - http://ecee.colorado.edu/~mcclurel/Philips_I2C_IO_Expanders_AN469_2.pdf

- src\devices\Tm1637\README.md
  - http://olimex.cl/website_MCI/static/documents/Datasheet_TM1637.pdf

- src\devices\Mpu6886\README.md
  - https://m5stack.oss-cn-shenzhen.aliyuncs.com/resource/docs/datasheet/core/MPU-6886-000193%2Bv1.1_GHIC_en.pdf

[broken-links.txt](https://github.com/dotnet/iot/files/9911802/broken-links.txt)

Certain paths can be ignored by adding the appropriate pattern to `.mlc-config.json` e.g.:

```
{
  "ignorePatterns": [
    {
      "pattern": "^http://example.net"
    }
  ]
}
```

**Comment by krwq** [comment]
@Jimmys20 I'll suggest create that config file and add each link with missing datasheet or similar (outside of our repo) individually and fix it separately. Once you're done let's open issue and point to that ignore file that those datasheets are missing and let's figure out path forward (possibly we should be copying those datasheets into our repo, not sure but we should figure out generic solution). In this PR let's focus on enabling the automatic checks and make sure links within the repo are correct

File: .mlc-config.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (krwq) [inline .mlc-config.json:1]: I'll suggest to match naming with: https://github.com/dotnet/iot/blob/main/.markdownlint.json

so something like `.markdown-link-checker.json`

**Comment by krwq** [comment]
Should the CI be showing link checker errors? I'm not seeing any

**Comment by Jimmys20** [comment]
The last broken links within repo are in https://github.com/dotnet/iot/blob/main/samples/bmp280-sensor-azure-iot-hub/Readme.md: `../.vscode/launch.json` and `../.vscode/tasks.json`. Any suggestion for those?

**Review by krwq — APPROVED** [review]
File an issue on the patterns we had to add to suppress errors and assuming there is no more errors LGTM and thank you for the effort!

**Comment by krwq** [comment]
Thanks again @Jimmys20 I'll file an issue shortly

**Comment by Jimmys20** [comment]
> Should the CI be showing link checker errors? I'm not seeing any

Is it possible that the automatic check should be applied in this file: https://github.com/dotnet/iot/blob/893ed04963166847e91ed5d85fe4e9ab109728a5/azure-pipelines.yml#L45 instead of `.vsts-ci.yml`? Or maybe in both? I'm not sure if the automatic check is working.

**Pull request merged**